### PR TITLE
Fix testing week-of-year extraction in `general-timezone-support-test`

### PR DESCRIPTION
**Pull request by calherries**
This PR fixes an issue where we were not testing week-of-year extraction by accident.

### CODE REVIEW

File: test/metabase/query_processor_test/timezones_test.clj [hunk 1]
```diff
@@ -248,7 +248,7 @@
       (for [i (range 366)]
         [(u.date/add start-date :day i)])]]))
 
-(defmethod driver/database-supports? [:driver ::extract-week-of-year-us]
+(defmethod driver/database-supports? [::driver/driver ::extract-week-of-year-us]
   [_driver _feature _database]
   true)
```
Comment (calherries) [inline test/metabase/query_processor_test/timezones_test.clj:253]: Before, this incorrectly returned false:
```
(mt/with-driver :h2 (driver/database-supports? :h2 ::extract-week-of-year-us (mt/db)))
```
Now it returns true.

Comment (crisptrutski) [reply test/metabase/query_processor_test/timezones_test.clj:253]: Given how easy this mistake was to make, should we not add a failing test that this fixes?

**Review by crisptrutski — APPROVED** [review]

**Comment by crisptrutski** [comment]
Good catch! How did you end up noticing this?

File: test/metabase/query_processor_test/timezones_test.clj (hunk 1 above)
Comment (calherries) [reply test/metabase/query_processor_test/timezones_test.clj:253]: I wouldn't, I think the mistake was made when we switched from `:driver` to `::driver/driver`. Now that the keyword is global I can't see how we could repeating that mistake.

**Comment by calherries** [comment]
Just stumbled across it. I was working on a test in the same namespace

**Pull request merged**